packages/mapviewer/src/service-workers.ts

```diff
@@ -46,7 +46,9 @@ if (!IS_TESTING_WITH_CYPRESS) {
             // exclude all api calls, as the service worker might interacts with those in a way
             // that can shut down the service from the user's perspective
             // (injecting the cached index.html file instead of providing the expected output)
-            denylist: [/^\/api\//,],
+            // excluding the embed legacy endpoint, as it stops the redirection to the
+            // `legacyEmbed` route and display map views instead of embed views
+            denylist: [/^\/api\//, / ^\/embed/,],
```

Comment:
I'd split the comment based on the line it belongs to
```javascript
    denylist: [
        // exclude all api calls, as the service worker might interacts with those in a way
        // that can shut down the service from the user's perspective
        // (injecting the cached index.html file instead of providing the expected output)
        /^\/api/,
        // excluding the embed legacy endpoint, as it stops the redirection to the
        // `legacyEmbed` route and display map views instead of embed views
        /^\/embed/,
    ],
``` 
careful of the extra space in the embed regex, don't know if that would behave like you intend it to with it